game-core/src/main/java/games/strategy/util/PointFileReaderWriter.java

```diff
@@ -188,6 +188,69 @@ public static void writeOneToMany(final OutputStream sink, final Map<String, Lis
     return mapping;
   }
 
+  /**
+   * Writes the specified one-to-many mapping between names and (points, overflowToLeft) to the specified stream.
+   *
+   * @param sink The stream to which the name-to-points mappings will be written.
+   * @param mapping The name-to-points mapping to be written.
+   *
+   * @throws IOException If an I/O error occurs while writing to the stream.
+   */
+  public static void writeOneToManyPlacements(final OutputStream sink,
+      final Map<String, Tuple<List<Point>, Boolean>> mapping)
+      throws IOException {
+    checkNotNull(sink);
+    checkNotNull(mapping);
+
+    final StringBuilder out = new StringBuilder();
+    final Iterator<String> keyIter = mapping.keySet().iterator();
+    while (keyIter.hasNext()) {
+      final String name = keyIter.next();
+      out.append(name).append(" ");
+      final Collection<Point> points = mapping.get(name).getFirst();
+      final boolean overflowToLeft = mapping.get(name).getSecond();
+      final Iterator<Point> pointIter = points.iterator();
+      while (pointIter.hasNext()) {
+        final Point point = pointIter.next();
+        out.append(" (").append(point.x).append(",").append(point.y).append(")");
+        if (pointIter.hasNext()) {
+          out.append(" ");
+        }
+      }
+      out.append(" | overflowToLeft=" + overflowToLeft);
+      if (keyIter.hasNext()) {
+        out.append("\r\n");
+      }
```

Comment:
This seems to be the only reason to use an iterator rather than a typical foreach loop.
Using Streams (as shown earlier) this additional check is uneccessary.
You might even want to put the "print point array" procedure into its own method to be able to separately unit-test it and to make using streams really easy.

Reply:
As a side note: A lot of code in this class looks very similar to each other.
Outsourcing methods will probably make this kind of duplication very obvious

Reply:
While there is some further refactoring that could be done, I think introducing streams would be better done for the entire class for consistency.